server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3383,12 +3383,6 @@ protected UserVm createVirtualMachine(DataCenter zone, ServiceOffering serviceOf
                     hostName = generateHostName(uuidName);
                 }
             }
-            // If global config vm.instancename.flag is set to true, then CS will set guest VM's name as it appears on the hypervisor, to its hostname.
-            // In case of VMware since VM name must be unique within a DC, check if VM with the same hostname already exists in the zone.
```

Comment:
Instead of removing the check, does it make sense to amend it to verify that with the updated naming convention -- there are no duplicate VMs by name?